### PR TITLE
drop pruned blocks asynchronously

### PR DESCRIPTION
**Pull request by msmouse**
### Description

helps with performance when blocks are large.

### Test Plan

tests (fixed)

### CODE REVIEW

File: execution/executor/src/components/block_tree/mod.rs [hunk 1]
```diff
@@ -250,8 +261,26 @@ impl BlockTree {
...
+        // guarantees that the drop will not happen in the current thread
+        let (tx, rx) = channel::<()>();
+        rayon::spawn(move || {
+            let _timeer = APTOS_EXECUTOR_OTHER_TIMERS_SECONDS
+                .with_label_values(&["drop_old_root"])
+                .start_timer();
+            drop(old_root);
+            // Error is ignored, since the caller might not care about dropping completion and
+            // has discarded the receiver already.
+            tx.send(()).ok();
```
Comment (zekun000) [inline execution/executor/src/components/block_tree/mod.rs:281]: you just want the join handle right?

Comment (msmouse) [reply execution/executor/src/components/block_tree/mod.rs:281]: `spawn()` doesn't return one.

**Review by areshand — APPROVED** [review]

File: execution/executor/src/components/block_tree/mod.rs (hunk 1 above)
Comment (msmouse) [reply execution/executor/src/components/block_tree/mod.rs:281]: Spawning a new thread will be slower in theory?

Comment (zekun000) [reply execution/executor/src/components/block_tree/mod.rs:281]: I guess it's probably faster than the channel lol

Comment (grao1991) [reply execution/executor/src/components/block_tree/mod.rs:281]: The cost of spawning a thread should be ~10us, not too bad.

**Review by grao1991 — APPROVED** [review]

File: execution/executor/src/components/block_tree/mod.rs (hunk 1 above)
Comment (zekun000) [reply execution/executor/src/components/block_tree/mod.rs:281]: just allocating the channel is not a fair comparison, you need to send something in a separate rayon task lol

Comment (msmouse) [reply execution/executor/src/components/block_tree/mod.rs:281]: Yeah but that's not important because it's async and noboday receive in production.

Comment (msmouse) [reply execution/executor/src/components/block_tree/mod.rs:281]: Ideally we have a global drop pool

Comment (zekun000) [reply execution/executor/src/components/block_tree/mod.rs:281]: it's just ugly but anyway spawning thread is much more expensive than i expected

Comment (msmouse) [reply execution/executor/src/components/block_tree/mod.rs:281]: landing ugly for now. maybe creating the drop pool "some time in the future"

**Pull request merged**